pkg/apis/extensions/v1beta1/types.go

```diff
@@ -286,6 +290,43 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentProgressing means the deployment is progressing.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentFailed means the deployment has failed its execution.
```

Comment:
Maybe a little more specific to explain what failed itself means in this context? Like, it could mean that the desired template can't be rolled out and won't be tried again while in this condition?